libs/testserver/http_server.go

```diff
@@ -0,0 +1,37 @@
+package testserver
+
+import (
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+// NewHttpServer creates a new httptest.Server that will respond with the same body that was sent to it
+// and will check if the headers match the expected headers.
+func NewHttpServer(t *testing.T, expectHeaders map[string]string) *httptest.Server {
```

Comment:
The `testserver` package already has another function that spawns a test server. It's better to inline this in the proxy library or maybe consolidate it with the other test server implementation.

Reply:
You mean fake_workspace server? I believe they have very different purposes though, I wouldn't consolidate them. But having test methods to start simple HTTP / Websocket servers might be still useful for some other tests. WDYT?